Added react hooks and components of practising

ndineshd opened the pull request:
This PR addresses the creation of class components and function components  with the help of useState and useEffect to update count values, mouseMove pointer value and auto count addresses in PR-#1.
## UI:-

![Screenshot 2023-10-04 181910](https://github.com/ndineshd/learning-react/assets/143182442/f89eb8de-31e0-45e1-b04a-6b1b0356a1c0)

![Screenshot 2023-10-04 181932](https://github.com/ndineshd/learning-react/assets/143182442/03d35519-e98c-4354-995d-623326b7c66c)
---
# 04-oct
Modified inline functions from buttons, Removed 'import React', Added components file to organise imports and exports in a single central file and Added 'IntervalHookCounter' to increase auto count.
## UI:-

![Screenshot 2023-10-04 182859](https://github.com/ndineshd/learning-react/assets/143182442/21aaf594-1a8c-40d5-a4ec-8f8db27e3dfb)

![Screenshot 2023-10-04 182934](https://github.com/ndineshd/learning-react/assets/143182442/6736435d-9195-4179-b663-2bbb2d65d094)

---
### 9e5555c style: Used shorthand margine property
 - Changed margine top and bottom to shorthand property
---
### 683d627 refactor: Rename button handle function
- Renamed the button handle function
- Removed the import react
---

### c5a84c7 chore: Organize component imports and exports
- Imported all components in a single file
- Exported all components from the central import/export file
---
### eaba0cb chore: Remove 'import React'
---
### dad3ef8 refactor: Remove inline function from button and 'import React'
- Removed the inline function from the button component
- Removed the 'import React'
---
### 83bf45d refactor: Remove inline function from button
---
### 0417dff refactor: Remove inline function from button, initialized value
- Removed inline functions from button
- Initialized direct value with 'useState'
---
### 63ccfc5 refactor: Remove inline function from buttons
---
### e869a50 refactor: Remove inline function from button
- Removed named export and 'import React'
---
### 8b15d24 refactor: Remove inline function from button
- Removed 'import React,
- Removed inline function from button
---
### d1fb4cf feat: Add functional component to increase auto count
- Used function component with 'useState'
- Used 'useEffect' to update count
---
### 9b08e6e refactor: Remove 'import React'
---
### f734d8a refactor: Remove 'import React'
---
### 68e13c7 chore: Consolidate component imports into a single file
- Removed all individual component imports
- Imported all components from a single central file

ndineshd committed:
feat: Add import

-Imported the reacthooks component to app.js

ndineshd committed:
feat: Add components and import components

-Imported components are UseState, HookCounter, HookPreState, ObjectUseState and ArrayUseState.

ndineshd committed:
feat: Add UseState component

-Added increment count by using class component
-Used constructor with properties
-Added onClick function to a button to increase the count value
-Exported the UseState function

ndineshd committed:
feat: Add HookCounter function

-Added increment count by using functional component.
-Used useState function with initial value 0.
-Added onClick function to a button to increase the count value.
-Exported the HookCounter function

ndineshd committed:
feat: Add HookPreState function

-Added increment, decrement and reset count by using functional component.
-Used useState function with initialCount 0.
-Added onClick functions to a button to increase,decrease and reset the count value.
-Exported the HookPreState function

ndineshd committed:
feat: Add ArrayUseState component

-Added Math.floor(Math.random()) to get automatic num
-Added UseState to store the num as an array
-Added function to button to add randon num
-Added map to print the list of random num

ndineshd committed:
feat: Add ObjectUseState component

-Added value and onChane attributes to input tags to read the event value and id
-Added UseState to store the current value with setter function
-Added spread operator to spread the values

ndineshd committed:
style: Add margin

-Added margin to top and bottom

ndineshd committed:
feat: Add components and import components

-Added components of UseEffect, UseEffectFunction, UseEffectMouse, FunctionMouse and MouseContainer
-Imported all the components
-Modified h2 to h1 tag

ndineshd committed:
feat: Add component to display count on page title

-Added class component with constructor to define count and initialised it to 0
-Added compoundDidMount and componentDidUpdate to add the updated count values to page title
-Define the function to a button to update and increase count values

ndineshd committed:
feat: Add functional component to update count value to page title

-Added function component and defined useState to update the count value
-Used UseEffect to update the count value on page title
-Added button to increase count value

ndineshd committed:
feat: Add component to print mouse move position

-Added class component with constructor to define x and y initialised to 0.
-Added logMousePosition and componentDidMount to update the x and y position values

ndineshd committed:
feat: Add function component to add and remove remove eventlistiner

-Added functional components to initialize the x and y
-Added logMousePosition to update the mouse position values
-Added useEffect to addeventlistener and returned to remove eventlistener

ndineshd committed:
feat: Add functional component to remove eventlistner

-Added functional component with useState to initialise display property
-Added function to button remove and add event listiner
-Added and imported FunctionMouse to add the mouse positions and useeffect to toogle button

ndineshd committed:
docs: Deployed

-Deployed learning-react with update the packages

yatensharma commented on src/App.css:
```diff
@@ -1,5 +1,7 @@
 .App {
   text-align: center;
+  margin-top: 40px;
+  margin-bottom: 80px;
```
use the shorthand notation for margins:

Like this: `margin: 40px 0 80px 0` also explore more shorthand like this `margin: 2px 4px`

yatensharma commented on src/ReactHooks/ReactHooks.js:
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
```
No need of importing React since you are using React-18

yatensharma commented on src/ReactHooks/ReactHooks.js:
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import UseState from './components/UseState'
+import HookCounter from './components/HookCounter'
+import HookPreState from './components/HookPreState'
+import ObjectUseState from './components/ObjectUseState'
+import ArrayUseState from './components/ArrayUseState'
+import UseEffect from './components/UseEffect'
+import UseEffectFunction from './components/UseEffectFunction'
+import UseEffectMouse from './components/UseEffectMouse'
+import FunctionMouse from './components/FunctionMouse'
+import MouseContainer from './components/MouseContainer'
```
Keep it in a single file and import it from there

yatensharma commented on src/ReactHooks/components/ArrayUseState.js:
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from 'react'
```
No need to import react at every place

yatensharma commented on src/ReactHooks/components/ArrayUseState.js:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <>
+    <h3>useState with Array</h3>
+    <button onClick={addItem}>Add a Number</button>
```
Make function name more clear something like `handleAddItem`

yatensharma commented on src/ReactHooks/components/ArrayUseState.js:
```diff
@@ -0,0 +1,27 @@
+            value: Math.floor(Math.random() * 10) + 1
+        }])
+    }
+  return (
```
Leave a line before return statement.

yatensharma commented on src/ReactHooks/components/ArrayUseState.js:
```diff
@@ -0,0 +1,27 @@
+  )
+}
+
+export default ArrayUseState
```
Read about default and named export and difference b/w them

yatensharma commented on src/ReactHooks/components/FunctionMouse.js:
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react'
```
Refer above comment

yatensharma commented on src/ReactHooks/components/HookCounter.js:
```diff
@@ -0,0 +1,13 @@
+
+function HookCounter() {
+    const [count, setCount] = useState(0 )
+  return (
```
Leave a new line

yatensharma commented on src/ReactHooks/components/HookCounter.js:
```diff
@@ -0,0 +1,13 @@
+  return (
+    <div>
+      <h3>using usestate in function</h3>
+      <button onClick={() => setCount(count + 1)} title='Click to increase the value'>Count {count}</button>
```
Remove the inline invocation of function

yatensharma commented on src/ReactHooks/components/HookPreState.js:
```diff
@@ -0,0 +1,25 @@
+    <div>
+        <h3>Hook Pre State using useState</h3>
+        <p>count: {count}</p>
+        <button onClick={() => setCount(initialCount)} title='Click to reset the value'>Reset</button>
+        <button onClick={() => setCount(prevCount => prevCount + 1)} title='click to increase the value'>Increment</button>
+        <button onClick={() => setCount(prevCount => prevCount - 1)} title='click to decrease the value'>Decrement</button>
```
Remove inline functions

yatensharma commented on src/ReactHooks/components/HookPreState.js:
```diff
@@ -0,0 +1,25 @@
+
+function HookPreState() {
+    const initialCount =0
+    const [count, setCount] = useState(initialCount)
```
directly provide `initialValue` to useState hook

yatensharma commented on src/ReactHooks/components/MouseContainer.js:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <div>
+        <h3>Removing and displaying event listeners</h3>
+      <button onClick={() => setDisplay(!display)}>Toggle display</button>
```
remove inline invocation

yatensharma commented on src/ReactHooks/components/ObjectUseState.js:
```diff
@@ -0,0 +1,27 @@
+
+    <form>
+
+        <input type='text' value={name.firstName} onChange={e => setName({...name, firstName: e.target.value})} />
+
+        <input type='text' value={name.lastName} onChange={e => setName({...name, lastName: e.target.value})} />
```
Refer previous comments.

yatensharma commented on src/ReactHooks/components/UseEffectFunction.js:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <div>
+        <h3>UseEffect with Function</h3>
+      <button onClick={() => setCount(count + 1)} title='Click and chect it on page title'>click {count} times</button>
```
refer prev comments

ndineshd committed:
style: Used shorthand margine property

-Changed margine top and bottom to shorthand property

ndineshd committed:
refactor: Rename button handle function

-Renamed the button handle function
-Removed the import react

ndineshd committed:
chore: Organize component imports  and exports

-Imported all components in a single file
-Exported all components from the central import/export file

ndineshd committed:
chore: Remove 'import React'

ndineshd committed:
refactor: Remove inline function from button and  'import React'

-Removed the inline function from the button component
-Removed the 'import React'

ndineshd committed:
refactor: Remove inline function from button

ndineshd committed:
refactor: Remove inline function from button, initialized value

- Removed inline functions from button
- Initialized direct value with 'useState'

ndineshd committed:
refactor: Remove inline function from buttons

ndineshd committed:
refactor: Remove inline function from button

- Removed named export and 'import React'

ndineshd committed:
refactor: Remove inline function from button

- Removed 'import React,
- Removed inline function from button

ndineshd committed:
feat: Add functional component to increase auto count

- Used function component with 'useState'
- Used 'useEffect' to update count

ndineshd committed:
refactor: Remove 'import React'

ndineshd committed:
refactor: Remove 'import React'

ndineshd committed:
chore: Consolidate component imports into a single file

- Removed all individual component imports
- Imported all components from a single central file
